cpp/src/reductions/collect_ops.cu

```diff
@@ -83,9 +76,15 @@ std::unique_ptr<scalar> merge_sets(lists_column_view const& col,
                                    rmm::cuda_stream_view stream,
                                    rmm::mr::device_memory_resource* mr)
 {
-  auto flatten_col = col.get_sliced_child(stream);
-  auto scalar      = std::make_unique<list_scalar>(flatten_col, true, stream, mr);
-  return drop_duplicates(*scalar, nulls_equal, nans_equal, stream, mr);
+  auto flatten_col    = col.get_sliced_child(stream);
+  auto distinct_table = detail::distinct(table_view{{flatten_col}},
+                                         std::vector<size_type>{0},
```

Comment:
No, they take `std::vector` because vector that will be passed down to `table_view::select` which requires `std::vector`. If we want to have span here, we must firstly modify `table_view::select` to take span.

Reply:
Ok, that's an entirely separate change. Please ignore.